userspace/libsinsp/gen_filter.cpp

```diff
@@ -166,6 +171,24 @@ bool gen_event_filter_expression::compare(gen_event *evt)
 	return res;
 }
 
+// This returns the union of all event types for all
+// filterchecks. Remember that the default is the empty set/no
+// specific event types, so that becomes restricted only when a
+// filtercheck returns a non-empty set.
```

Comment:
So, `evt.type != open and evt.type != openat` will end up with a set of all syscalls, right? How much do we care?

Reply:
The driver for this PR is within falco, so it can quickly match an event type against a set of rules that are applicable for that event type. This used to be done within falco itself as it parsed the rule condition, but we're changing falco so it's not explicitly tied to specific kinds of events like syscall events and doesn't have to download a set of "known" event type values like "open", "openat", etc.

This matching can have false positives (e.g. linking an event type to a rule when the rule condition won't actually match the event type), but cannot have false negatives (e.g. not linking an event type to a rule condition that could evaluate to true for that event type).

This matching is only an initial test, though--the rule conditions still are fully evaluated for any rules linked to the event type. So for this specific filter expression, events with all types would pass the initial test, but only events that really are not open and not openat would evaluate to true for the filter.

Hopefully this provides enough context.

Reply:
Hmm. How about `evt.type=connect or fd.type=ipv4`? That will end up with the set containing just the `connect` events, right?

Reply:
Right, but for that filter you would only want connect events right? So I think it works.

Reply:
With addl changes these filters should work, and should return the exact set of event types too.